foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/actions/CountAction.java

```diff
@@ -106,7 +107,12 @@ public ActionResponse execute(CountRequest parameter) throws FoxtrotException {
             }
 
             try {
-                SearchResponse response = query.execute().actionGet();
+                SearchResponse response;
+                if(isCountQueryTimeBounded()) {
```

Comment:
"If else" doesn't seems good. Let's have very high timeout if user is not providing we will use that. So, no condition and else part.